### PR TITLE
Crud for news on react

### PR DESCRIPTION
**Pull request by Swooty18**
[Trello Task](https://trello.com/_____)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [x] DB indexes added if needed
- [x] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI
изображение

### CODE REVIEW

File: app/controllers/api/v1/news_controller.rb [hunk 1]
```diff
@@ -1,10 +1,55 @@
 class Api::V1::NewsController < Api::ApiController
+  before_action :set_news_by_role
+  before_action :set_post, only: %i[show update destroy]
   def index
+    authorize :news
     @news = current_user.osbb.news.page params[:page]
   end
 
   def show
     @post = News.find(params[:id])
```
Comment (chu4) [inline app/controllers/api/v1/news_controller.rb:10]: you have this in set_post

Comment (Swooty18) [reply app/controllers/api/v1/news_controller.rb]: refactor

**Review by Ukhanskyi — APPROVED** [review]

**Review by mmarusyk — APPROVED** [review]

**Review by Andrii-Kozak — APPROVED** [review]

File: app/controllers/api/v1/news_controller.rb [hunk 2]
```diff
@@ -1,10 +1,52 @@
...
+    authorize :news
     @news = current_user.osbb.news.page params[:page]
   end
 
-  def show
-    @post = News.find(params[:id])
+  def show; end
+
+  def update
+    authorize :news
```
Comment (chu4) [inline app/controllers/api/v1/news_controller.rb:12]: you already have authorize in set_post

File: app/controllers/api/v1/news_controller.rb [hunk 3]
```diff
@@ -1,10 +1,52 @@
...
+    @post.osbb_id = current_user.osbb_id
+    if @post.save
+      render json: @post
+    else
+      render json: @post.errors
+    end
+  end
+
+  def destroy
+    News.destroy(params[:id])
```
Comment (chu4) [inline app/controllers/api/v1/news_controller.rb:31]: why do you need before action set_post for this one?

**Review by chu4 — APPROVED** [review]

**Review by IvanDanyliv — APPROVED** [review]

**Pull request merged**